### PR TITLE
Feat(isolation tab): created isolation tab

### PR DESCRIPTION
**Pull request by roymaniv**

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+                                האם יש סימפטומים:
+                            </b>
+                        </Typography>
+                    </Grid>
+                    <Grid item xs={10}>
+                        <Toggle
+                            value={hasSymptoms}
+                            onChange={hasSymptomsToggle}
+                        />
+                    </Grid>
```
Comment (sandymash) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx:206]: should probably divide into subform components

Comment (sandymash) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx:202]: each comment (like {/* האם יש סימפטומים */}) should be a component by itself

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+        id: 4,
+        name: 'אחר',
+    },
+
+];
+
+const hospitals: string[] = ['שיבא', 'איכילוב', 'אסף הרופא'];
+
+let selectedSymptoms: string[] = [];
+let selectedBackgroundIllnesses: string[] = [];
```
Comment (sandymash) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx:68]: why isnt it a state?

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetailsStyles.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        display: 'flex',
+        flexDirection: 'row',
+        marginTop: '2vh',
+    },
+    smallGrid: {
+        width: '40vw'
+    },
+    isolationAddress: {
+        display: 'flex',
+        flexDirection: 'row',
```
Comment (sandymash) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetailsStyles.ts:17]: its row by default tbh

Comment (sandymash) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetailsStyles.ts:17]: redundant

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+                                        <Grid item xs={6}>
+                                            <CustomCheckbox
+                                                checkboxElements={[{
+                                                    key: backgroundIllness.id,
+                                                    value: backgroundIllnessesList.find((chosenBackgroundIllness) => chosenBackgroundIllness.name === backgroundIllness.name),
+                                                    text: backgroundIllness.name,
+                                                    onChange: () => {
+                                                        handleBackgroundIllnessCheck(backgroundIllness)
+                                                        updateClinicalDetails(ctxt as ClinicalDetailsDataAndSet, ClinicalDetailsFields.BACKGROUND_ILLNESSES, selectedBackgroundIllnesses)
+                                                    }
```
Comment (sandymash) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx:283]: move to named func probably

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+                        />
+                    </Grid>
+                    <Grid item xs={2}>
+                    </Grid>
+                    <Grid item xs={10}>
+                        <Collapse in={hasSymptoms}>
+                            <div className={classes.dates}>
+                                <DatePick
+                                    datePickerType='date'
+                                    value={(!isUnkonwnDateChecked && ctxt.clinicalDetailsData?.symptomsStartDate !== null) ? format(ctxt.clinicalDetailsData?.symptomsStartDate as Date, 'yyyy-MM-dd') : 'yyyy-MM-dd'}
```
Comment (sandymash) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx:210]: should be const or state or calculated field

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+        if (selectedBackgroundIllnesses.find(checkedName => checkedName === backgroundIllness.name)) {
+            selectedBackgroundIllnesses = selectedBackgroundIllnesses.filter((checkedBackgroundIllness) => checkedBackgroundIllness !== backgroundIllness.name);
+        } else {
+            selectedBackgroundIllnesses.push(backgroundIllness.name);
+        };
+    };
+
+    return (
+        <ClinicalDetailsDataContextConsumer>
+            {ctxt => (
```
[the remark this answers is not in the document]
Comment (roymaniv) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx:111]: למה צריך אבל useContext?

[the remark this answers is not in the document]
Comment (sandymash) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx:111]: an then ctxt prop in funcs can be removed as well

File: client/src/commons/DatePick/DatePickStyles.ts [hunk 7]
```diff
@@ -11,6 +11,9 @@ export const useStyles = makeStyles({
         paddingBottom: '1.5vh'
     },
     textField: {
-        width: '9.5vw',
+        width: '12vw',
+    },
+    text: {
+        marginTop: '2.3%',
```
Comment (RoeiRom) [inline client/src/commons/DatePick/DatePickStyles.ts:17]: Why % and not vh?

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx [hunk 8]
```diff
@@ -0,0 +1,361 @@
...
+        if (selectedBackgroundIllnesses.find(checkedName => checkedName === backgroundIllness.name)) {
+            setSelectedBackgroundIllnesses(selectedBackgroundIllnesses.filter((checkedBackgroundIllness) => checkedBackgroundIllness !== backgroundIllness.name));
+        } else {
+            selectedBackgroundIllnesses.push(backgroundIllness.name);
+        };
+
+        updateClinicalDetails(ClinicalDetailsFields.BACKGROUND_ILLNESSES, selectedBackgroundIllnesses)
+    };
+
+    const context = React.useContext(clinicalDetailsDataContext);
```
Comment (RoeiRom) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx:112]: Move up, isn’t it creating new context?

File: client/src/components/App/Content/InvestigationForm/TabManagement/TabManagementStyles.ts [hunk 9]
```diff
@@ -4,8 +4,9 @@ const useStyles = makeStyles({
     card: {
         height: '67vh',
     },
-    displayedTab: {
-        height: '61vh'
+    displyedTab: {
+        height: '61vh',
+        overflow: 'scroll'
```
Comment (RoeiRom) [inline client/src/components/App/Content/InvestigationForm/TabManagement/TabManagementStyles.ts:9]: Use auto

**Review by RoeiRom — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx (hunk 8 above)
Comment (roymaniv) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx:112]: it is

Comment (RoeiRom) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx]: so how is it updating the context you provided

Comment (roymaniv) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetails.tsx]: its the same context used in investigationForm

**Review by RoeiRom — APPROVED** [review]

**Review by OrelKaravani — APPROVED** [review]
@sandymash  approved

**Pull request merged**